kratos/containers/flags.h

```diff
@@ -238,6 +238,12 @@ class KRATOS_API(KRATOS_CORE) Flags
         mFlags = BlockType();
     }
 
+    static const Flags Unset(Flags ThisFlag)
+    {
+        Flags unset_this_flag(ThisFlag);
+        unset_this_flag.mFlags = !ThisFlag.mFlags;
+        return  unset_this_flag;
+    }
```

Comment:
hmmm, I'm not sure about being a static member. In fact, I am not sure if the behavior is correct. The static one should only mark the define and set the flags to zero. The no static version should unset the given flags bits in the this->mflags and this->mDefined.